kubernetes/metadata/replicaset.go

```diff
@@ -94,3 +98,32 @@ func (rs *replicaset) GenerateFromName(name string, opts ...FieldOptions) mapstr
 
 	return nil
 }
+
+// RemoveUnnecessaryReplicaSetData removes all data from a ReplicaSet resource, except what we need to compute
+// Pod metadata. This function works for both ReplicaSet and PartialObjectMetadata.
+func RemoveUnnecessaryReplicaSetData(obj interface{}) (interface{}, error) {
+	switch old := obj.(type) {
+	case *appsv1.ReplicaSet:
+		transformed := &appsv1.ReplicaSet{
+			ObjectMeta: kubernetes.ObjectMeta{
+				Name:            old.GetName(),
+				Namespace:       old.GetNamespace(),
+				OwnerReferences: old.GetOwnerReferences(),
+				ResourceVersion: old.GetResourceVersion(),
+			},
+		}
+		return transformed, nil
+	case *metav1.PartialObjectMetadata:
+		transformed := &metav1.PartialObjectMetadata{
+			ObjectMeta: kubernetes.ObjectMeta{
+				Name:            old.GetName(),
+				Namespace:       old.GetNamespace(),
+				OwnerReferences: old.GetOwnerReferences(),
+				ResourceVersion: old.GetResourceVersion(),
+			},
+		}
+		return transformed, nil
+	default:
+		return nil, fmt.Errorf("obj of type %v neither a ReplicaSet nor a PartialObjectMetadata", reflect.TypeOf(obj))
```

Comment:
question: cant we use in the fmt `%T` and thus not have to import reflect here?! 

quoting from [here](https://pkg.go.dev/fmt#hdr-Printing)
> %T a Go-syntax representation of the type of the value